docs/resources/lakehouse_monitor.md

```diff
@@ -0,0 +1,135 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_lakehouse_monitor Resource
+
+This resource allows you to manage [Lakehouse Monitors](https://docs.databricks.com/en/lakehouse-monitoring/index.html) in Databricks. 
+
+A `databricks_lakehouse_monitor` is attached to a [databricks_sql_table](sql_table.md) and can be of type timeseries, snapshot or inference. 
+
+## Example Usage
+
+```hcl
+
+resource "databricks_catalog" "sandbox" {
+  name    = "sandbox"
+  comment = "this catalog is managed by terraform"
+  properties = {
+    purpose = "testing"
+  }
+}
+
+resource "databricks_schema" "things" {
+    catalog_name = databricks_catalog.sandbox.id
+    name         = "things"
+    comment      = "this database is managed by terraform"
+    properties = {
+      kind = "various"
+    }
+  }
+  
+resource "databricks_table" "myTestTable" {
+    catalog_name = "main"
+    schema_name = databricks_schema.things.name
+    name = "bar"
+    table_type = "MANAGED"
+    data_source_format = "DELTA"
+
+    column {
+        name      = "timestamp"
+        position  = 1
+        type_name = "INT"
+        type_text = "int"
+        type_json = "{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}"
+    }
+}
+
+resource "databricks_lakehouse_monitor" "testTimeseriesMonitor" {
+    table_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}.${databricks_table.myTestTable.name}"
+    assets_dir = "/Shared/provider-test/databricks_lakehouse_monitoring/${databricks_table.myTestTable.name}"
+    output_schema_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}"
+    time_series  {
+        granularities = ["1 hour"]
+        timestamp_col = "timestamp"
+    }
+}
+```
+
+### Inference Monitor
+
+```hcl
+resource "databricks_lakehouse_monitor" "testMonitorInference" {
+    table_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}.${databricks_table.myTestTable.name}"
+    assets_dir = "/Shared/provider-test/databricks_lakehouse_monitoring/${databricks_table.myTestTable.name}"
+    output_schema_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}"
+    inference_log  {
+        granularities = ["1 hour"]
+        timestamp_col = "timestamp"
+        prediction_col = "prediction"
+        model_id_col = "model_id"
+        problem_type = "PROBLEM_TYPE_REGRESSION"
+    } 
+}
+```
+### Snapshot Monitor
+```hcl
+resource "databricks_lakehouse_monitor" "testMonitorInference" {
+    table_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}.${databricks_table.myTestTable.name}"
+    assets_dir = "/Shared/provider-test/databricks_lakehouse_monitoring/${databricks_table.myTestTable.name}"
+    output_schema_name = "${databricks_catalog.sandbox.name}.${databricks_schema.things.name}"
+    snapshot  {} 
```

Comment:
Looks like a bit weird syntax, would it be better to expose this as a boolean instead and then convert in the resource implementation?

Reply:
We debated about this. We changed the Go SDK to use an empty struct in place of `any`. This is also the the example expected in the Go SDK, so thought it would be clear. Miles also did not want too many changes between Go SDK structs and the terraform input structs. 

Also in the future if there are any snapshot relevant parameters the team introduces, we dont need additional changes in the terraform provider as we are already using the struct.